plugin/pkg/auth/authorizer/node/node_authorizer.go

```diff
@@ -231,6 +231,23 @@ func (r *NodeAuthorizer) authorize(nodeName string, startingType vertexType, att
 	return authorizer.DecisionAllow, "", nil
 }
 
+// authorizeServiceAccount authorizes
+// - "list", "watch" requests to serviceaccounts when KubeletServiceAccountTokenForCredentialProviders feature is enabled
+// - "create" requests to serviceaccounts 'token' subresource of pods running on a node
+func (r *NodeAuthorizer) authorizeServiceAccount(nodeName string, startingType vertexType, attrs authorizer.Attributes) (authorizer.Decision, string, error) {
+	verb := attrs.GetVerb()
+	switch verb {
+	case "list", "watch":
+		if !r.features.Enabled(features.KubeletServiceAccountTokenForCredentialProviders) {
+			klog.V(2).Infof("NODE DENY: '%s' %#v", nodeName, attrs)
+			return authorizer.DecisionNoOpinion, "not allowed to list or watch service accounts", nil
```

Comment:
should this say why?

Reply:
Not sure if should say why! The service account manager in kubelet is only initialized if the `KubeletServiceAccountTokenForCredentialProviders` feature flag is enabled and that's the only manager that requires the `list` and `watch`. I don't think we want to necessarily correlate this feature flag to getting permissions for `list, watch`. LMK what you think!